### DISCUSSION TITLE
problem with data migration from v. 1.2

### DISCUSSION THREAD

**Discussion by rekombination** [kimai/kimai#4799] [On-Premise: Installation & Updates] [answered]
### Describe the issue

Hello,

I am currently facing challenges with migrating data from Kimai version 1.2, 
which was originally transferred to the server via FTP, to the latest version of Kimai. 
To ensure compatibility with the database schema from version 1.2, 
I installed v. 1.2 using Git and Composer, which required some effort.

I exported the SQLite database data using the DB Browser. However, when attempting to import this 
data into the fresh MySQL database via phpMyAdmin, I encountered an error.

```
MySQL said: Documentation

#1064 - You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '"kimai2_users" ("id","username","username_canonical","email","email_canonical","' at line 1

```

The first line of the exported sql file looks like this: 

`INSERT INTO "kimai2_users" ("id","username","username_canonical","email","email_canonical","enabled","salt","password","last_login","confirmation_token","password_requested_at","roles","alias","registration_date","title","avatar","api_token") VALUES (1,'admin','admin','contact@example.com','contact@mexample.com',1,'YUBgLcpMf/ceWeIevrSjCNyYuUfn754WCV00SHUHv74','$argon2id$v=19$m=65536,t=4,p=1$Skw1cjN6ZnFOQ0tTcnJleQ$+FTZ4DijLjGy4B7JSDeNHbLzVRNT66oGsd/axfi7UG0','2024-04-11 20:13:43',NULL,NULL,'a:2:{i:0;s:16:"ROLE_SUPER_ADMIN";i:1;s:10:"ROLE_ADMIN";}',NULL,'2019-09-07 18:37:33',NULL,NULL,NULL);`

I would greatly appreciate your help in resolving this problem. 
Thank you in advance for your support!

Best regards

### I already tried

- [X] I've read and searched [the documentation](https://www.kimai.org/documentation/).
- [X] I've searched for similar issues in this repository.
- [X] I've searched for similar issues in [the discussions](https://github.com/kimai/kimai/discussions).

### Kimai version

1.2

### How do you run Kimai?

Shared-Hosting

### Which PHP version are you using?

7.4

**Comment by kevinpapst** [comment]
You are 3 years too late ... but all you need to know is in this blog post: 
https://www.kimai.org/blog/2021/sqlite-and-ftp-support-removed/

**Comment by rekombination** [comment]
Yes, I'm aware that the version I'm dealing with is quite outdated. I've based my attempts to migrate the data on the methods described in the blog post you mentioned. Unfortunately, these steps haven't resolved the issue for me. In theory, it should work as described in the blog, right?

Could you possibly offer any additional advice  that might help resolve the database import problem?

**Reply by kevinpapst** [reply, marked as answer]
Yes, it works. 
I migrated several sqlite installations with the ruby method. 
Haven't tried the other ones. 

But afaik Mysql uses backticks ` for column names, not "

**Reply by rekombination** [reply]
thank you very much for the hint. i didn't know....
this way the import works.

best.
